common/src/main/java/edu/kit/kastel/mcse/ardoco/core/common/util/SimilarityUtils.java

```diff
@@ -252,19 +263,137 @@ private static boolean checkRecommendedInstanceForSelection(ModelInstance instan
         ImmutableList<String> longestNameSplit = Lists.immutable.of(CommonUtilities.splitCases(instance.getFullName()).split(" "));
         ImmutableList<String> recommendedInstanceNameList = Lists.immutable.with(ri.getName());
         if (areWordsSimilar(instance.getFullName(), ri.getName()) || SimilarityUtils.areWordsOfListsSimilar(instanceNames, recommendedInstanceNameList,
-                similarity) || SimilarityUtils.areWordsOfListsSimilar(longestNameSplit, recommendedInstanceNameList, similarity)) {
+                similarity) || SimilarityUtils.areWordsOfListsSimilar(longestNameSplit, recommendedInstanceNameList, similarity) || 1.0 * similarEntriesOfList(
+                        instanceNames, recommendedInstanceNameList) / Math.max(instanceNames.size(), recommendedInstanceNameList
+                                .size()) >= similarity || 1.0 * similarEntriesOfList(longestNameSplit, recommendedInstanceNameList) / Math.max(instanceNames
+                                        .size(), recommendedInstanceNameList.size()) >= similarity) {
```

Comment:
This is really complex. I would suggest to create boolean variables for the cases that are checked here; e.g.,

boolean similarNames = areWordsSimilar(instance.getFullName(), ri.getName());
boolean similarInstanceNames = ....